Assets/MixedRealityToolkit/Interfaces/SceneSystem/IMixedRealitySceneSystem.cs

```diff
@@ -0,0 +1,245 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine.SceneManagement;
+
+namespace Microsoft.MixedReality.Toolkit.SceneSystem
+{
+    /// <summary>
+    /// Interface for managing scenes in Unity.
+    /// Scenes are divided into three categories: Manager, Lighting and Content.
+    ///
+    /// The Manager scene is loaded first and remains loaded for the duration of the app.
+    /// Only one Manager scene is ever loaded, and no scene operation will ever unload it.
+    ///
+    /// The Lighting scene is a largely empty scene which controls lighting settings.
+    /// Ambient lighting, skybox, sun direction, etc. A default lighting scene is loaded on initialization.
+    /// After that the active lighting scene may be changed at any time via SetLightingScene.
+    /// Only one lighting scene can ever be loaded at a time.
+    ///
+    /// Content scenes are everything else. These can be loaded and unloaded at will in any combination.
+    ///
+    /// The scene actions provided improve on unity's SceneManagement events by ensuring that scenes
+    /// are considered valid before the action is invoked.
+    /// </summary>
+    public interface IMixedRealitySceneSystem : IMixedRealityEventSystem, IMixedRealityEventSource, IMixedRealityDataProvider
+    {
+        #region Actions
+
+        /// <summary>
+        /// Called just before a set of content scenes is loaded.
+        /// Includes names of all scenes about to be loaded.
+        /// </summary>
+        Action<IEnumerable<string>> OnWillLoadContent { get; set; }
```

Comment:
Should this (and similar) be OnBegin* instead of OnWill*?

Reply:
Not sure. I'm (roughly) following the OnWillRenderObject pattern but it may not be appropriate here.